sdk/core/azure_core/src/options/retry.rs

```diff
@@ -0,0 +1,208 @@
+use crate::policies::{ExponentialRetryPolicy, FixedRetryPolicy, NoRetryPolicy, Policy};
+use crate::RetryPolicy;
+use std::fmt::Debug;
+use std::sync::Arc;
+use std::time::Duration;
+
+/// The algorithm to apply when calculating the delay between retry attempts.
+#[derive(Clone)]
+enum RetryMode {
+    /// Retry attempts will delay based on a back-off strategy,
+    /// where each attempt will increase the duration that it waits before retrying.
+    ///
+    /// This is the default.
+    Exponential(ExponentialRetryOptions),
+
+    /// Retry attempts happen at fixed intervals; each delay is a consistent duration.
+    Fixed(FixedRetryOptions),
+
+    /// A custom retry policy
+    Custom(Arc<dyn Policy>),
+
+    /// Do not retry attempts.
+    None,
+}
+
+impl Debug for RetryMode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RetryMode::Exponential(o) => write!(f, "Exponential({o:?})"),
+            RetryMode::Fixed(o) => write!(f, "Fixed({o:?})"),
+            RetryMode::Custom(_) => write!(f, "Custom"),
+            RetryMode::None => write!(f, "None"),
+        }
+    }
+}
+
+impl Default for RetryMode {
+    fn default() -> Self {
+        RetryMode::Exponential(ExponentialRetryOptions::default())
+    }
+}
+
+/// Specify how retries should behave.
+///
+/// Note that not all requests can be retried. These options will only be used
+/// when a retry is attempted.
+///
+/// The default is an exponential retry policy using the default `ExponentialRetryOptions`.
+#[derive(Clone, Debug, Default)]
+pub struct RetryOptions {
+    /// The algorithm to use for calculating retry delays.
+    mode: RetryMode,
+}
+
+impl RetryOptions {
+    /// A retry strategy where attempts happen at intervals that get exponentially longer with each retry.
```

Comment:
Does filling in the rest of the doccomment (parameter description for instance) make sense now?

Reply:
I don't understand. Could you clarify what you mean?

Reply:
It was a generic comment related to a lot of the parts of this file. For instance, the to_policy method has no doccomment, there are methods with a 1 line description and parameters without descriptions, etc.

I assume that all this will need to be filled in at some point or other, but I was wondering if now was the right time, or if it was going to be handled in a later PR.

Reply:
Thanks for clarifying offline. Retries will be changing a bit when I refactor and I was planning on doing it then to see what survives as-is (or close). I opened #1709 to track.